docs/en/integrations/index.asciidoc

```diff
@@ -0,0 +1,915 @@
+include::{docs-root}/shared/versions/stack/{source_branch}.asciidoc[]
+include::{docs-root}/shared/attributes.asciidoc[]
+
+:doctype: book
+:forum: https://discuss.elastic.co/
+
+:package-spec-repo: https://github.com/elastic/package-spec
+
+// Access package spec files with {package-spec-root}
+:package-spec-version: 1
+:package-spec: {package-spec-root}/versions/{package-spec-version}
+
+= Integrations Developer Guide
+
```

Comment:
I think we can also copy the content from https://github.com/elastic/integrations/pull/1029 to this documentation as a set of generic guidelines. @ycombinator @mtojek thoughts?